clang/test/CodeGenOpenCL/vector-to-scalar-coercion.cl

```diff
@@ -0,0 +1,23 @@
+// RUN: %clang_cc1 -x cl -triple i686-unknown-unknown   -fopencl-force-vector-abi %s -O0 -emit-llvm -o - | FileCheck %s --check-prefix NOCOER
+// RUN: %clang_cc1 -x cl -triple x86_64-unknown-unknown -fopencl-force-vector-abi %s -O0 -emit-llvm -o - | FileCheck %s --check-prefix NOCOER
+// RUN: %clang_cc1 -x cl -triple i686-unknown-unknown   %s -O0 -emit-llvm -o - | FileCheck %s --check-prefix COER32CL
+// RUN: %clang_cc1 -x cl -triple x86_64-unknown-unknown %s -O0 -emit-llvm -o - | FileCheck %s --check-prefix COER64CL
+// RUN: %clang_cc1 -x c  -triple i686-unknown-unknown   %s -O0 -emit-llvm -o - | FileCheck %s --check-prefix COER32
+// RUN: %clang_cc1 -x c  -triple x86_64-unknown-unknown %s -O0 -emit-llvm -o - | FileCheck %s --check-prefix COER64
+
+typedef unsigned short ushort;
+typedef ushort ushort4 __attribute__((ext_vector_type(4)));
+
+typedef unsigned long ulong;
+typedef ulong ulong4 __attribute__((ext_vector_type(4)));
+
+ulong4 __attribute__((const)) __attribute__((overloadable)) convert_ulong4_rte(ushort4 x)
+{
+  return 1;
+}
+
+// NOCOER:     define {{.*}}<4 x i64> @_Z18convert_ulong4_rteDv4_t(<4 x i16> noundef %{{.*}})
+// COER32CL:   define {{.*}}<4 x i64> @_Z18convert_ulong4_rteDv4_t(i64 noundef %{{.*}})
+// COER64CL:   define {{.*}}<4 x i64> @_Z18convert_ulong4_rteDv4_t(double noundef %{{.*}})
```

Comment:
Is 'double' the right type here?

Reply:
this is not related to this PR.

If we add ElementType->isSpecificBuiltinType(BuiltinType::UShort) to https://github.com/intel/llvm/blob/aa69e4d9b86c9f0e1cb109d7f6870584f3328908/clang/lib/CodeGen/TargetInfo.cpp#L2974, the 'double' will be changed to 'i64'. Does it make sense to add all integer types to that place?

Reply:
I think we should, otherwise I am concerned the representation will be incorrect.  Is something like isIntegerType or isIntegralOrEnumerationType() more appropriate there instead of checking each integral type kind there?

@erichkeane, could you please give your input here?   I am really not sure.

Reply:
ping @erichkeane . thanks

Reply:
This 'double' type issue isn't related to this PR. @premanandrao could you please approve this PR if there is no other issues? This PR blocks the pre-commit build of https://github.com/intel/llvm/pull/8216

Reply:
Thanks @erichkeane!
@hewj03, I think TargetInfo.cpp should be fixed to not have this converted to a double.   Please create an issue to fix that in a separate PR if it won't be done as part of this.   I don't think we should lose track of that issue.  And add a FIXME to this test where we see the possibly incorrect 'double' types.

Reply:
> Please create an issue to fix that in a separate PR if it won't be done as part of this.

Issue https://github.com/intel/llvm/issues/8347 is created for the 'double' type.